### DISCUSSION TITLE
"Live featured project" was enabled for all users by default

### DISCUSSION THREAD

**Discussion by WorldLanguages** [ScratchAddons/ScratchAddons#686] [Discussion]
We could set it to `"enabledByDefault": false` from now, but I don't think it's a good solution. Some users might have liked the feature.

IMO we should keep it as a default, but require user interaction to load the project, which is the biggest problem - we're loading the whole Scratch VM and stuff each time our users go to a profile, by default. Instead, adding a little "play project inline" button hurts no one, since there's already a lot of empty space after "Featured Projects" (in most languages, of course).

My suggestion is:
1. Keep the addon enabled by default
2. Rename "auto-play projects" to "auto-load projects", but keep the setting ID to preserve previous user choices.

There would be no auto-play option anymore. If the user has auto-load disabled, the project should autoplay, since the user clicked a "play project" button. If the user has it enabled, it should auto-load, but *not* autoplay. Seriously, I don't think anyone wants to have sound coming out out of nowhere. Also, browsers might not like sound playing without user interaction.
If a user really liked the feature, and enabled auto-play, then after the update they'll have auto-load enabled, which is not exactly the same, but at least we haven't trashed their settings.
If a user disabled the addon, we'll enable it for them again, making sure auto-load is disabled.
This would be for v1.3.1

What do you think?
cc @Hans5958

**Comment by Hans5958** [comment]
Nah, disable it by default. It is unnecessary to add another button upon turning on an addon because it brokes the philosophy of SA of (a) users needing to turning addons manually within the first install/update and (b) (most) addons work straight away upon turning it on without user interaction. 

The ones that are impacted are the v1.3.0 users, and that's where the happy users from, and when it is disabled within the next update, they will need to turn it on manually, and I think this is the best case.

EDIT: My conclusion is a request to merge #684, because I quite pissed off with this, my second mistake.

P/S: I could go farther by suggesting having the next update to turn off the live featured project after SA is updated, so they need to turn it on manually, and it aligns with the philosophy.

**Reply by WorldLanguages** [reply]
I see your point, but I think you're forgetting this was a native Scratch feature back then :P

**Reply by Hans5958** [reply]
Also, another reason. Everyone who uses Scratch knows that only the thumbnail that will be shown, so some users only do the thumbnail, leaving the project empty. I think there is plenty of people who did this, and of course, makes the addon quite useless, even after loading the alternative player (because there's nothing to play).

I will try to make an option to turn it off, so it reverts the changes when it fails, leaving the thumbnail alone, but this is a big reason why I didn't want this to be enabled by default. 

Also, about the fact that this is an earlier feature on Scratch. Times changing, so does the demographic of the users, and most, if not all, users didn't know that it was an original feature at all.

**Comment by williamhorning** [comment]
I think that it should load the project by default, but add a small flag icon above the featured project so the user can play it. This still messes up my animated profile banner. But I could just add the code to the project and redo the thumbnail.

**Reply by Hans5958** [reply]
OT, but you can do that by turning off the auto-play, and show the menu bar. You can just click anywhere on the screen and it plays the project, ust like our ol' Scratch project.
